File: example/src/main/java/com/jiangcm/baselib/guideView/component/SimpleComponent.java
package com.jiangcm.baselib.guideView.component;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.Toast;

import androidx.constraintlayout.widget.ConstraintLayout;

import com.jiangcm.base.widght.guideview.Component;
import com.jiangcm.baselib.R;

/**
 * Created by binIoter on 16/6/17.
 */
public class SimpleComponent implements Component {

  @Override public View getView(LayoutInflater inflater) {

    ConstraintLayout ll = (ConstraintLayout) inflater.inflate(R.layout.layer_frends, null);
    ll.setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View view) {
        Toast.makeText(view.getContext(), "引导层被点击了", Toast.LENGTH_SHORT).show();
      }
    });
    ll.post(() -> {
      Log.v("thss","ll "+ll.getWidth())     ;
      Log.v("thss","ll "+ll.getHeight())     ;
    });
    return ll;
  }

  @Override public int getAnchor() {
    return Component.ANCHOR_OVER;
  }

  @Override public int getFitPosition() {
    return Component.FIT_CENTER;
  }

  @Override public int getXOffset() {
    return 0;
  }

  @Override public int getYOffset() {
    return 10;
  }
}

File: example/src/main/java/com/jiangcm/baselib/ui/guideView/component/SimpleComponent.kt
package com.jiangcm.baselib.ui.guideView.component

import android.view.LayoutInflater
import android.view.View
import androidx.constraintlayout.widget.ConstraintLayout
import com.jiangcm.base.widght.guideview.Component
import com.jiangcm.baselib.R

class SimpleComponent : Component {

    var listener: (() -> Unit?)? = null

    fun setOnListener(listener: () -> Unit?): SimpleComponent {
        this.listener = listener
        return this
    }

    override fun getView(inflater: LayoutInflater?): View {
        val ll = inflater?.inflate(R.layout.layer_frends, null) as ConstraintLayout
        ll.setOnClickListener { view ->
            listener?.invoke()
        }
        return ll
    }

    override fun getAnchor(): Int = Component.ANCHOR_OVER
    override fun getFitPosition(): Int = Component.FIT_CENTER
    override fun getXOffset(): Int = 0
    override fun getYOffset(): Int = 0
}